addons/web/static/src/js/owl_compatibility.js

```diff
@@ -494,6 +492,19 @@ odoo.define('web.OwlCompatibility', function () {
             }
             parentChildren.push(this);
         }
+        /**
+         * Stores mount target and position at first mount. That way, when updating
+         * while out of DOM, we know where and how to remount.
+         * @see update()
+         * @override
+         */
+        async mount(target, options) {
+            if (options && options.position === 'self') {
+                throw new Error(`Unsupported position: "self" is not allowed for wrapper component.`);
```

Comment:
i would have done this in update instead, no? You can mount it with position self, as soon as your widget isn't detached and re-inserted later on. What do you think?

Reply:
@aab-odoo 
this would be a very strong assumption to make. i.e. you need the legacy widget to never have been detached from the DOM to assert that you'd need the Error only in update.